fastestimator/estimator.py

```diff
@@ -181,32 +183,41 @@ def _prepare_traces(self) -> None:
         brought back from the GPU to CPU for use in traces. This also prints a warning if no model saver trace is
         detected.
         """
+        no_save_warning = True
         modes = self.pipeline.get_modes()
         loss_keys = self.network.get_loss_keys()
         if "train" in modes:
             self.traces.insert(0, TrainEssential(loss_keys=loss_keys))
         if "eval" in modes and loss_keys.issubset(self.network.get_all_output_keys("eval", self.system.total_epochs)):
-            self.traces.insert(1, EvalEssential(loss_keys=loss_keys, monitor_names=self.monitor_names))
+            self.traces.insert(1, EvalEssential(monitor_names=self.monitor_names | loss_keys))
         if self.system.log_steps is not None:
             self.traces.append(Logger(extra_log_keys=self.monitor_names))
         for mode in modes:
             trace_inputs = set()
             # '*' is a reserved key for traces to indicate that they want to receive all available output
             trace_outputs = {"*"}
             for trace in self.traces:
-                if not trace.mode or mode in trace.mode:
-                    trace_inputs.update(trace.inputs)
-                    trace_outputs.update(trace.outputs)
+                if isinstance(trace, Scheduler):
+                    for trace_ in trace.get_all_values():
+                        if trace_:
+                            trace_.system = self.system
+                            if isinstance(trace, (ModelSaver, BestModelSaver)):
+                                no_save_warning = False
+                            if not trace_.mode or mode in trace_.mode:
+                                trace_inputs.update(trace_.inputs)
```

Comment:
this doesn't allow us to catch issues where key exists from pipeline only in first 10 epochs, but trace wants it on epoch 20

Reply:
true, this issue is in previous code too.    I guess this is the best we can do for now.